### PR TITLE
Add provision to create and delete DBSubnetGroups for RDS integration test

### PR DESCRIPTION
**Pull request by akankshakumari393**
## Change Overview

This PR modifies RDS Postgres and Aurora apps to create and delete DBSubnet group. Also adds some utility functions related to that.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/app/rds_aurora_mysql.go [hunk 1]
```diff
@@ -138,6 +141,44 @@ func (a *RDSAuroraMySQLDB) Install(ctx context.Context, namespace string) error
...
+	for _, subnet := range resp.Subnets {
+		subnetIDs = append(subnetIDs, *subnet.SubnetId)
+	}
+	// create a subnetgroup with subnets in the VPC
+	subnetGroup, err := rdsCli.CreateDBSubnetGroup(ctx, fmt.Sprintf("%s-subnetgroup", a.name), "kanister-test-subnet-group", subnetIDs)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to create subnet group")
+	}
+	a.dbSubnetGroup = *subnetGroup.DBSubnetGroup.DBSubnetGroupName
+
```
Comment (viveksinghggits) [inline pkg/app/rds_aurora_mysql.go:181]: Let's extract this newly added code in a new function/method in the same source file and call that from here. This method has become too lengthy.

File: pkg/app/rds_aurora_mysql.go [hunk 2]
```diff
@@ -328,6 +364,21 @@ func (a *RDSAuroraMySQLDB) Uninstall(ctx context.Context) error {
 		return errors.Wrap(err, "Failed to create ec2 client.")
 	}
 
+	// Delete dbSubnetGroup
```
Comment (viveksinghggits) [inline pkg/app/rds_aurora_mysql.go:367]:
```suggestion
```

File: pkg/app/rds_postgres.go [hunk 3]
```diff
@@ -141,6 +149,39 @@ func (pdb *RDSPostgresDB) Install(ctx context.Context, ns string) error {
...
+	for _, subnet := range resp.Subnets {
+		subnetIDs = append(subnetIDs, *subnet.SubnetId)
+	}
+	// create a subnetgroup with subnets in the VPC
+	subnetGroup, err := rdsCli.CreateDBSubnetGroup(ctx, fmt.Sprintf("%s-subnetgroup", pdb.name), "kanister-test-subnet-group", subnetIDs)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to create subnet group")
+	}
+	pdb.dbSubnetGroup = *subnetGroup.DBSubnetGroup.DBSubnetGroupName
+
```
Comment (viveksinghggits) [inline pkg/app/rds_postgres.go:184]: Same comment has above, and is there possibility to re-use these between both the apps?

Comment (akankshakumari393) [reply pkg/app/rds_postgres.go:175]: CreateDBSubnetGroup should be there in install method, but I have tried refactor the other things, PTAL.

File: pkg/app/utils.go [hunk 4]
```diff
@@ -94,3 +96,21 @@ func bastionDebugWorkloadSpec(ctx context.Context, name string, image string, na
 		},
 	}
 }
+
+// getVpcIdForRDSInstance gets the VPC_ID from env if set or from the default VPC
+func getVpcIdForRDSInstance(ctx context.Context, ec2Cli *ec2.EC2) (string, error) {
```
Comment (PrasadG193) [inline pkg/app/utils.go:101]: Words in names that are initialisms or acronyms have a consistent case. (https://github.com/golang/go/wiki/CodeReviewComments#initialisms)
```suggestion
func getVPCIDForRDSInstance(ctx context.Context, ec2Cli *ec2.EC2) (string, error) {
```

File: pkg/app/rds_aurora_mysql.go [hunk 5]
```diff
@@ -138,25 +140,49 @@ func (a *RDSAuroraMySQLDB) Install(ctx context.Context, namespace string) error
...
+		return errors.Wrapf(err, "Failed to describe subnets")
+	}
+
+	// Extract subnet IDs from the response
+	var subnetIDs []string
+	for _, subnet := range resp.Subnets {
+		subnetIDs = append(subnetIDs, *subnet.SubnetId)
+	}
+	// create a subnetgroup with subnets in the VPC
+	subnetGroup, err := rdsCli.CreateDBSubnetGroup(ctx, fmt.Sprintf("%s-subnetgroup", a.name), "kanister-test-subnet-group", subnetIDs)
```
Comment (PrasadG193) [inline pkg/app/rds_aurora_mysql.go:166]: Lets declare `kanister-test-subnet-group` as const

File: pkg/app/rds_aurora_mysql.go [hunk 6]
```diff
@@ -328,6 +354,20 @@ func (a *RDSAuroraMySQLDB) Uninstall(ctx context.Context) error {
...
 	}
 
+	log.Info().Print("Deleting db subnet group.", field.M{"app": a.name})
+	_, err = rdsCli.DeleteDBSubnetGroup(ctx, a.dbSubnetGroup)
+	if err != nil {
+		// If the subnet group does not exist, ignore the error and return
+		if err, ok := err.(awserr.Error); ok {
+			switch err.Code() {
+			case awsrds.ErrCodeDBSubnetGroupNotFoundFault:
+				log.Info().Print("Subnet Group Does not exist: ErrCodeDBSubnetGroupNotFoundFault.", field.M{"app": a.name, "name": a.dbSubnetGroup})
```
Comment (PrasadG193) [inline pkg/app/rds_aurora_mysql.go:364]: missing return?

File: pkg/app/rds_postgres.go [hunk 7]
```diff
@@ -141,23 +149,41 @@ func (pdb *RDSPostgresDB) Install(ctx context.Context, ns string) error {
...
+	var subnetIDs []string
+	for _, subnet := range resp.Subnets {
+		subnetIDs = append(subnetIDs, *subnet.SubnetId)
+	}
+	// create a subnetgroup with subnets in the VPC
+	subnetGroup, err := rdsCli.CreateDBSubnetGroup(ctx, fmt.Sprintf("%s-subnetgroup", pdb.name), "kanister-test-subnet-group", subnetIDs)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to create subnet group")
+	}
+	pdb.dbSubnetGroup = *subnetGroup.DBSubnetGroup.DBSubnetGroupName
```
Comment (PrasadG193) [inline pkg/app/rds_postgres.go:174]: I see this as a common code in aurora and rds files. Can we move it to a helper method?

File: pkg/app/utils.go [hunk 8]
```diff
@@ -94,3 +96,21 @@ func bastionDebugWorkloadSpec(ctx context.Context, name string, image string, na
...
+		defaultVpc, err := ec2Cli.DescribeDefaultVpc(ctx)
+		if err != nil {
+			return "", err
+		}
+		if len(defaultVpc.Vpcs) == 0 {
+			return "", fmt.Errorf("No default VPC found")
+		}
+		return *defaultVpc.Vpcs[0].VpcId, nil
+	}
+	return vpcID, nil
```
Comment (PrasadG193) [inline pkg/app/utils.go:115]:
```suggestion
	// VPCId is not provided, use Default VPC
	if vpcID != "" {
	        return vpcID, nil
	}
	defaultVpc, err := ec2Cli.DescribeDefaultVpc(ctx)
	if err != nil {
		return "", err
	}
	if len(defaultVpc.Vpcs) == 0 {
		return "", fmt.Errorf("No default VPC found")
	}
	return *defaultVpc.Vpcs[0].VpcId, nil
```
[nit] for better readability

File: pkg/app/rds_postgres.go [hunk 9]
```diff
@@ -130,6 +132,12 @@ func (pdb *RDSPostgresDB) Install(ctx context.Context, ns string) error {
 		return err
 	}
 
+	// Create rds client
+	rdsCli, err := rds.NewClient(ctx, awsConfig, region)
+	if err != nil {
+		return err
+	}
+
```
Comment (viveksinghggits) [inline pkg/app/rds_postgres.go:140]:
```suggestion
	rdsCli, err := rds.NewClient(ctx, awsConfig, region)
	if err != nil {
		return err
	}

```

File: pkg/app/rds_aurora_mysql.go (hunk 6 above)
Comment (akankshakumari393) [reply pkg/app/rds_aurora_mysql.go:363]: yeah, because we don't want to return in case its not found.

File: pkg/app/utils.go [hunk 10]
```diff
@@ -94,3 +98,44 @@ func bastionDebugWorkloadSpec(ctx context.Context, name string, image string, na
 		},
 	}
 }
+
+// getVpcIdForRDSInstance gets the VPC_ID from env if set or from the default VPC
+func getVPCIDForRDSInstance(ctx context.Context, ec2Cli *ec2.EC2) (string, error) {
```
Comment (viveksinghggits) [inline pkg/app/utils.go:103]:
```suggestion
// vpcIDForRDSInstance gets the VPC ID from env var `VPC_ID` if set, or from the default VPC
func vpcIDForRDSInstance(ctx context.Context, ec2Cli *ec2.EC2) (string, error) {
```

File: pkg/app/utils.go [hunk 11]
```diff
@@ -94,3 +98,44 @@ func bastionDebugWorkloadSpec(ctx context.Context, name string, image string, na
...
+	}
+	if len(defaultVpc.Vpcs) == 0 {
+		return "", fmt.Errorf("No default VPC found")
+	}
+	return *defaultVpc.Vpcs[0].VpcId, nil
+}
+
+// getDBSubnetGroup gets the DBSubnetGroup based on VPCID
+func getDBSubnetGroup(ctx context.Context, ec2Cli *ec2.EC2, rdsCli *rds.RDS, vpcID, name, subnetGroupDescription string) (string, error) {
+	// describe subnets in the VPC
```
Comment (viveksinghggits) [inline pkg/app/utils.go:122]:
```suggestion
// dbSubnetGroup gets the DB subnet group based on VPC ID
func dbSubnetGroup(ctx context.Context, ec2Cli *ec2.EC2, rdsCli *rds.RDS, vpcID, name, subnetGroupDescription string) (string, error) {
```

**Review by viveksinghggits — CHANGES_REQUESTED** [review]
LGTM

File: pkg/app/utils.go [hunk 12]
```diff
@@ -94,3 +98,44 @@ func bastionDebugWorkloadSpec(ctx context.Context, name string, image string, na
...
+	if err != nil {
+		return "", err
+	}
+	if len(defaultVpc.Vpcs) == 0 {
+		return "", fmt.Errorf("No default VPC found")
+	}
+	return *defaultVpc.Vpcs[0].VpcId, nil
+}
+
+// dbSubnetGroup gets the DBSubnetGroup based on VPCID
```
Comment (viveksinghggits) [inline pkg/app/utils.go:120]:
```suggestion
// dbSubnetGroup gets the DBSubnetGroup based on VPC ID
```

**Review by viveksinghggits — APPROVED** [review]

**Pull request merged**